Add locks

gurux13 opened the pull request:
Currently the only way to wait for data is in a busy loop.
This adds locking mechanisms to the Raspberry Pi, when compiled with WITH_LOCKS.

Locke commented:
Note: if this should be merged my alternative PR #87 is probably obsolete / could be closed as won't fix with a reference to this one.

1technophile commented:
This PR currently break the CI:
https://travis-ci.org/github/sui77/rc-switch

In particular focus should be put also on the Arduino and ESP behaviours

gurux13 commented:
Fixed. I don't use this library for Arduino builds, so it's a little complicated for me to test on Arduino.
This PR does not support Arduino, so I've added a check that the locks are used only on a RPi.
If you find that this works on other platforms, please feel free to add proper preprocessor checks.
